lib/active_model/serializer.rb

```diff
@@ -59,17 +59,11 @@ class << self
     end
 
     # @api private
-    def self.serializer_lookup_chain_for(klass)
-      chain = []
-
-      resource_class_name = klass.name.demodulize
-      resource_namespace = klass.name.deconstantize
-      serializer_class_name = "#{resource_class_name}Serializer"
-
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
-      chain.push("#{resource_namespace}::#{serializer_class_name}")
-
-      chain
+    def self.serializer_lookup_chain_for(klass, controller_class = nil)
+      lookups = ActiveModelSerializers.config.serializer_lookup_chain
+      Array[*lookups].flat_map do |lookup|
+        lookup.call(klass, self, controller_class)
```

Comment:
ref [ActionController::Serialization in 0.9](https://github.com/rails-api/active_model_serializers/pull/1436#issuecomment-171823264)

```ruby

    def namespace_for_serializer
      @namespace_for_serializer ||= self.class.parent unless self.class.parent == Object
    end

    def default_serializer(resource)
      options = {}.tap do |o|
        o[:namespace] = namespace_for_serializer if namespace_for_serializer
      end

      ActiveModel::Serializer.serializer_for(resource, options)
    end

    def default_serializer_options
      {}
    end

    def serialization_scope
      _serialization_scope = self.class._serialization_scope
      send(_serialization_scope) if _serialization_scope && respond_to?(_serialization_scope, true)
    end

    def build_json_serializer(resource, options = {})
      options = default_serializer_options.merge(options)
      @namespace_for_serializer = options.fetch(:namespace, nil)

      if serializer = options.fetch(:serializer, default_serializer(resource))
        options[:scope] = serialization_scope unless options.has_key?(:scope)

        if resource.respond_to?(:to_ary)
          options[:resource_name] = controller_name 
          options[:namespace] = namespace_for_serializer if namespace_for_serializer
        end

        serializer.new(resource, options)
      end
```